### PR TITLE
feat: add design token file to component generation

### PR DESCRIPTION
**Pull request by kpanot**
## Proposed change

Add generation of the Design token file with the component when the package is added to the project

## Related issues

- :rocket: Feature #1249

### CODE REVIEW

File: packages/@o3r/core/schematics/rule-factories/component/design-token.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import type { Rule } from '@angular-devkit/schematics';
+import type { NgGenerateComponentSchematicsSchema } from '../../component/schema';
+import { askQuestionsToGetRulesOrThrowIfPackageNotAvailable } from './common';
+
+export const getAddDesignTokenRules = (
+  designTokenath: string,
```
Comment (matthieu-crouzet) [inline packages/@o3r/core/schematics/rule-factories/component/design-token.ts:6]: small typo
```suggestion
  designTokenPath: string,
```

File: packages/@o3r/core/schematics/rule-factories/component/design-token.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  stylePath: string,
+  options: Pick<NgGenerateComponentSchematicsSchema, 'useOtterDesignToken'>
+): Rule => askQuestionsToGetRulesOrThrowIfPackageNotAvailable(
+  designTokenath,
```
Comment (matthieu-crouzet) [inline packages/@o3r/core/schematics/rule-factories/component/design-token.ts:10]:
```suggestion
  designTokenPath,
```

File: packages/@o3r/core/schematics/component/presenter/index.ts [hunk 3]
```diff
@@ -83,6 +84,7 @@ function ngGenerateComponentPresenterFn(options: NgGenerateComponentSchematicsSc
     const o3rSpecPath = path.posix.join(componentDestination, `${properties.name}.spec.ts`);
     const ngStylePath = path.posix.join(componentDestination, `${properties.name}.component.scss`);
     const o3rStylePath = path.posix.join(componentDestination, `${properties.name}.style.scss`);
+    const o3rDesignTokenPath = path.posix.join(componentDestination, `${properties.name}.theme.json`);
```
Comment (matthieu-crouzet) [inline packages/@o3r/core/schematics/component/presenter/index.ts:87]: There https://github.com/AmadeusITGroup/otter/pull/1250/files#diff-e6eff8f06e5a25d2030aff90e52c72a8f88b91dc39f0495809963938b1518edeR21
You expect the file extension to be `.design-token.json`

File: packages/@o3r/design/schematics/design-token-to-component/index.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import { basename, dirname, relative } from 'node:path';
+import type { NgAddDesignTokenSchematicsSchema } from './schema';
+
+/**
+ * Add Design Token to an existing component
+ * @param options
+ */
+export function ngAddDesignTokenFn(options: NgAddDesignTokenSchematicsSchema): Rule {
+
+  const name = basename(options.path, '.design-token.json');
```
Comment (matthieu-crouzet) [inline packages/@o3r/design/schematics/design-token-to-component/index.ts:21]: Maybe we should specify in the doc that file should follow this naming convention
else they will have issue :)

Comment (matthieu-crouzet) [reply packages/@o3r/design/schematics/design-token-to-component/index.ts:21]: Or even better we can set a pattern validator on the schema

**Review by fpaul-1A — DISMISSED** [review]

File: packages/@o3r/core/schematics/component/presenter/index.ts [hunk 5]
```diff
@@ -83,6 +84,7 @@ function ngGenerateComponentPresenterFn(options: NgGenerateComponentSchematicsSc
     const o3rSpecPath = path.posix.join(componentDestination, `${properties.name}.spec.ts`);
     const ngStylePath = path.posix.join(componentDestination, `${properties.name}.component.scss`);
     const o3rStylePath = path.posix.join(componentDestination, `${properties.name}.style.scss`);
+    const o3rDesignTokenPath = path.posix.join(componentDestination, `${properties.name}.them.json`);
```
Comment (matthieu-crouzet) [inline packages/@o3r/core/schematics/component/presenter/index.ts:87]:
```suggestion
    const o3rDesignTokenPath = path.posix.join(componentDestination, `${properties.name}.theme.json`);
```

**Review by matthieu-crouzet — DISMISSED** [review]

**Review by matthieu-crouzet — APPROVED** [review]

**Review by fpaul-1A — APPROVED** [review]

**Pull request merged**